updater/lib/dependabot/analyse_files_command.rb

```diff
@@ -0,0 +1,152 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "base64"
+require "json"
+require "dependabot/base_command"
+require "dependabot/dependency_snapshot"
+require "dependabot/errors"
+require "dependabot/opentelemetry"
+require "dependabot/updater"
+require "github_api/dependency_submission"
+
+module Dependabot
+  class AnalyseFilesCommand < BaseCommand
+    extend T::Sig
+
+    # TODO(brrygrdn): Change label to analyse_files_error?
+    #
+    # It feels odd to return update_files_error, but Dependabot's backend service does a lot of categorisation
```

Comment:
Does this categorization happen in `-api`?

Reply:
It does, yes, these values are sorted into categories somewhat similar to HTTP 4xx, 5xx to distinguish cases where there is a problem with something user controlled vs a service problem.

For this new job type, all file parsing related error_types will be shared with updates, we just won't throw update-time error_types and might have a single new 'can't submit the payload' error to add.